python/src/fragment.rs

```diff
@@ -50,6 +53,24 @@ impl FileFragment {
         format!("LanceFileFragment(id={})", self.fragment.id())
     }
 
+    #[staticmethod]
+    #[pyo3(signature = (dataset_uri, datafile_uri, fragment_id))]
+    fn create_from_file(
+        dataset_uri: &str,
+        datafile_uri: &str,
+        fragment_id: usize,
+    ) -> PyResult<FragmentMetadata> {
+        let rt = Runtime::new()?;
+        let dataset = rt.block_on(async { Dataset::open(dataset_uri).await.unwrap() });
```

Comment:
Perhaps instead we should have the caller pass in the table schema? Right now this calls `Dataset::open` twice: once here, and then once again within `LanceFragment::create_from_file`. If a caller is doing this in a loop for many fragments, it would be more efficient if they just got the schema up front and passed it into each call.

Reply:
sounds reasonable